### PR TITLE
Add GOALS file, to define 1.0.

### PR DESCRIPTION
**Pull request by georgebrock**
What's left to do before we can call this 1.0?

### CODE REVIEW

**Comment by georgebrock** [comment]
This was inspired by a Slack conversation about versioning, and https://github.com/thoughtbot/terrapin/blob/master/GOALS

A lot of these things are described in more detail in GitHub issues, so maybe a GitHub Milestone would be a better place to put this?

File: GOALS [hunk 1]
```diff
@@ -0,0 +1,7 @@
+gitsh hits 1.0 when:
+
+[x] it provides a useful interactive shell environment.
+[x] it runs scripts.
+[ ] it has the kinds of features expected by advanced shell users.
```
Comment (tabfugnic) [inline GOALS]: This seems a bit vague. Should this have examples?

File: GOALS [hunk 2]
```diff
@@ -0,0 +1,7 @@
...
+[ ] its tab completion system is at least as good as Git's bash integration.
+[ ] it has a smooth install process on popular versions of macOS, BSD and GNU/Linux.
```
Comment (tabfugnic) [inline GOALS]: Happy to help with this!

File: GOALS [hunk 3]
```diff
@@ -0,0 +1,7 @@
+gitsh hits 1.0 when:
+
+[x] it provides a useful interactive shell environment.
+[x] it runs scripts.
```
Comment (tabfugnic) [inline GOALS]: While this is done, should this be a bit more descriptive?

**Comment by mike-burns** [comment]
Yeah or maybe a GH Issue Project?

In a Slack thread I listed these advantages:

- give everyone an indication of what it means for them to use a sub-1.0 project.
- give me a clear indication of when I've hit 1.0, so I won't run into perfectionism stopping me.
- help me get to 1.0 ASAP, so we can get normal semver going.

File: GOALS (hunk 1 above)
Comment (georgebrock) [reply GOALS]: I was thinking of these features: #321, #325, #145, #100, #67, and maybe also #303, #41, #35

Not sure if it's redundant to list them out here and in GitHub issues? I could also imagine someone suggesting a very reasonable feature that fits in this list and seems like it should be in 1.0, but maybe I should resist that kind of feature creep.

**Comment by georgebrock** [comment]
After the conversations here about repeating issues in a GOALS file, I set up a couple of issue milestones: https://github.com/thoughtbot/gitsh/milestones

I think these do the same job, although maybe the should be referenced in the contributing guide.

**Comment by mike-burns** [comment]
Oh that's great. LGTM. I guess you'd close this?

**Comment by georgebrock** [comment]
Yep, closing this in favour of GitHub milestones.

**Pull request closed without merging**